src/main/java/org/jabref/logic/importer/QueryConverter.java

```diff
@@ -0,0 +1,48 @@
+package org.jabref.logic.importer;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jabref.logic.importer.fetcher.ComplexSearchQuery;
+
+import org.apache.lucene.index.Term;
+import org.apache.lucene.queryparser.flexible.core.QueryNodeException;
+import org.apache.lucene.queryparser.flexible.standard.StandardQueryParser;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.QueryVisitor;
+
+/**
+ * This class converts a query string written in lucene syntax into a complex search query.
+ *
+ * For simplicity this is limited to fielded data and the boolean AND operator.
+ */
+public class QueryConverter {
```

Comment:
I would call this `QueryParser`. I guess parser is more common for conversion from string to some object (its also the name used in Lucene).